src/Components/Facility/FacilityHome.tsx

```diff
@@ -163,6 +187,7 @@ export const FacilityHome = ({ facilityId }: Props) => {
         open={editCoverImage}
         onSave={() => {
           facilityFetch();
+          fetchImage();
           setCoverImageEdited(true);
```

Comment:
Can we wait till the fetch image is done (in the loading state itself of the image edit modal) before running facility fetch?

Reply:
> Can we wait till the fetch image is done (in the loading state itself of the image edit modal) before running facility fetch?

Sure, so something like this? 

<img width="269" alt="image" src="https://github.com/user-attachments/assets/b5cf6839-139a-4971-894d-b959713fc173">

Reply:
yes, but ensure the loading state is shown in the image edit modal till it completes.